wire-ios-sync-engine/Source/SessionManager/SessionManager+Backup.swift

```diff
@@ -30,71 +31,67 @@ extension SessionManager {
 
     // MARK: - Export
 
-    public enum BackupError: Error {
-        case notAuthenticated
-        case noActiveAccount
-        case compressionError
-        case invalidFileExtension
-        case keyCreationFailed
-        case decryptionError
-        case unknown
-    }
-
-    public func backupActiveAccount(password: String) throws -> URL {
+    public func backupActiveAccount(password: String, completion: @escaping (Result<URL, Error>) -> Void) {
         guard
             let userId = accountManager.selectedAccount?.userIdentifier,
             let clientId = activeUserSession?.selfUserClient?.remoteIdentifier,
             let handle = activeUserSession.flatMap(ZMUser.selfUser)?.handle,
             let activeUserSession
         else {
-            throw BackupError.noActiveAccount
+            return completion(.failure(CreateLegacyBackupError.noActiveAccount))
         }
 
-        do {
-            let backupInfo = try CoreDataStack.backupLocalStorage(
-                accountIdentifier: userId,
-                clientIdentifier: clientId,
-                applicationContainer: sharedContainerURL,
-                dispatchGroup: dispatchGroup,
-                databaseKey: activeUserSession.managedObjectContext.databaseKey
-            )
-
-            return try SessionManager.handle(
-                result: .success(backupInfo),
-                password: password,
-                accountId: userId,
-                dispatchGroup: dispatchGroup,
-                handle: handle
-            )
-        } catch {
-            activeUserSession.analyticsEventTracker?.trackEvent(.Backup.exportFailed)
-            throw error
-        }
+        CoreDataStack.backupLocalStorage(
```

Comment:
Also here the changes are mostly reverting the epic branch changes to match develop.